arch/posix/CMakeLists.txt

```diff
@@ -16,25 +16,80 @@ endif()
 
 zephyr_compile_options(
   ${ARCH_FLAG}
-  -include ${ZEPHYR_BASE}/arch/posix/include/posix_cheats.h
   )
 
-# @Intent: Obtain compiler specific flags for no freestanding compilation
-zephyr_compile_options($<TARGET_PROPERTY:compiler,hosted>)
+if (CONFIG_NATIVE_APPLICATION)
+  zephyr_compile_options(
+    -include ${ZEPHYR_BASE}/arch/posix/include/posix_cheats.h
+  )
+elseif (CONFIG_NATIVE_LIBRARY)
+  if (CONFIG_64BIT)
+    # By default the native simulator builds and links a 32bits (x86) executable
+    target_link_options(native_simulator INTERFACE "-m64")
+    target_compile_options(native_simulator INTERFACE "-m64")
```

Comment:
This will not work on Linux/Aarch64 machines (for the most part). IIRC, there is some existing logic for that which should be used.

Reply:
Thanks @cfriedt, this if has been moved into the same if/else for Linux.Aarch64.